sklearn/manifold/t_sne.py

```diff
@@ -388,10 +388,10 @@ def _gradient_descent(objective, p0, it, n_iter, objective_error=None,
         new_error, grad = objective(p, *args, **kwargs)
         grad_norm = linalg.norm(grad)
 
-        inc = update * grad >= 0.0
+        inc = update * grad < 0.0
         dec = np.invert(inc)
-        gains[inc] += 0.05
-        gains[dec] *= 0.95
+        gains[inc] += 0.2
+        gains[dec] *= 0.8
         np.clip(gains, min_gain, np.inf)
```

Comment:
I just noticed that this line is completely useless. It should be

    np.clip(gains, min_gain, np.inf, out=gains)

Do we add it to this PR?